src/main/java/uk/gov/objecthash/DictValue.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.objecthash;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DictValue implements ObjectHashable {
+    private final Map<String, ObjectHashable> data;
+
+    public DictValue() {
+        this.data = new HashMap<>();
+    }
+    
+    public DictValue(Map<String, ObjectHashable> data) {
+        this.data = data;
+    }
+    
+    public byte[] digest() {
+        MessageDigest sha256 = ObjectHashable.sha256Instance();
+        
+        List<byte[]> hashValues = new ArrayList<>();
+        for (Map.Entry<String, ObjectHashable> entry : this.data.entrySet()) {
+            ObjectHashable value = entry.getValue();
+            ObjectHashable key = new StringValue(entry.getKey());
+            if (value != null) {
+                hashValues.add(concatEntryHashes(key.digest(), value.digest()));
+            }
+        }
+        hashValues.sort((a, b) -> compareBytes(a, b));
+        sha256.update(ObjectHashable.DICT_TAG.getBytes(StandardCharsets.UTF_8));
+        hashValues.stream().forEach(bytes -> sha256.update(bytes));
```

Comment:
Can be simplified to `(sha256::update)`

Reply:
I don't think `stream()` is needed here, i.e. can be simplified to `hashValues.forEach`